src/pages/settings/Payments/PaymentsPage/BasePaymentsPage.js

```diff
@@ -33,141 +31,92 @@ import ConfirmContent from '../../../../components/ConfirmContent';
-                shouldResetPaymentMethodData = true;
-            } else if (this.state.selectedPaymentMethodType === CONST.PAYMENT_METHODS.DEBIT_CARD && _.isEmpty(this.props.cardList[this.state.methodID])) {
-                shouldResetPaymentMethodData = true;
-            } else if (this.state.selectedPaymentMethodType === CONST.PAYMENT_METHODS.PAYPAL && this.props.payPalMeData !== prevProps.payPalMeData && _.isEmpty(this.props.payPalMeData)) {
-                shouldResetPaymentMethodData = true;
-            }
-            if (shouldResetPaymentMethodData) {
-                // Close corresponding selected payment method modals which are open
-                if (this.state.shouldShowDefaultDeleteMenu) {
-                    this.hideDefaultDeleteMenu();
-                } else if (this.state.shouldShowPasswordPrompt) {
-                    this.hidePasswordPrompt();
-                }
-            }
-        }
-
-        // previously online OR currently offline, skip fetch
-        if (!prevProps.network.isOffline || this.props.network.isOffline) {
-            return;
+import useLocalize from '../../../../hooks/useLocalize';
+import useWindowDimensions from '../../../../hooks/useWindowDimensions';
+
+function BasePaymentsPage(props) {
+    const {translate} = useLocalize();
+    const {isSmallScreenWidth, windowWidth} = useWindowDimensions();
+    const [shouldShowAddPaymentMenu, setShouldShowAddPaymentMenu] = useState(false);
+    const [shouldShowDefaultDeleteMenu, setShouldShowDefaultDeleteMenu] = useState(false);
+    const [showPassword, setShowPassword] = useState({
+        shouldShowPasswordPrompt: false,
+        passwordButtonText: '',
+    });
+    const [shouldShowLoadingSpinner, setShouldShowLoadingSpinner] = useState(false);
+    const [paymentMethod, setPaymentMethod] = useState({
+        isSelectedPaymentMethodDefault: false,
+        selectedPaymentMethod: {},
+        formattedSelectedPaymentMethod: {
+            title: '',
+        },
+        methodID: null,
+        selectedPaymentMethodType: null,
+    });
+    const [anchorPosition, setAnchorPosition] = useState({
+        anchorPositionHorizontal: 0,
+        anchorPositionVertical: 0,
+        anchorPositionTop: 0,
+        anchorPositionRight: 0,
+    });
+    const [addPaymentMethodButton, setAddPaymentMethodButton] = useState(null);
+    const [showConfirmDeleteContent, setShowConfirmDeleteContent] = useState(false);
+
+    const updateShouldShowLoadingSpinner = useCallback(() => {
+        // In order to prevent a loop, only update state of the spinner if there is a change
+        const showLoadingSpinner = props.isLoadingPaymentMethods || false;
+        if (showLoadingSpinner !== shouldShowLoadingSpinner) {
+            setShouldShowLoadingSpinner(props.isLoadingPaymentMethods && !props.network.isOffline);
         }
+    }, [props.isLoadingPaymentMethods, props.network.isOffline, shouldShowLoadingSpinner]);
 
-        this.fetchData();
-    }
```

Comment:
This fetch Data is not handled it should happen on reconnect

Reply:
Refer this https://github.com/Expensify/App/pull/22412#issuecomment-1638786387

Reply:
Hi @Santhosh-Sellavel 
Fixed reconnect issue.